lnx-engine/search-index/src/schema.rs

```diff
@@ -0,0 +1,538 @@
+use std::iter::FromIterator;
+
+use anyhow::{anyhow, Error, Result};
+use hashbrown::{HashMap, HashSet};
+use serde::{Deserialize, Serialize};
+use tantivy::schema::{
+    Cardinality,
+    FacetOptions,
+    Field,
+    FieldType,
+    IndexRecordOption,
+    IntOptions,
+    Schema,
+    SchemaBuilder,
+    TextFieldIndexing,
+    TextOptions,
+    FAST,
+    INDEXED,
+    STORED,
+};
+use tantivy::Score;
+
+use crate::helpers::{Calculated, Validate};
+
+pub static PRIMARY_KEY: &str = "_id";
+
+fn default_to_true() -> bool {
+    true
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SchemaContext {
+    /// The index's fields.
+    ///
+    /// Document entries will be implicitly converted into the index's schema types
+    /// if they need to be.
+    fields: HashMap<String, FieldDeclaration>,
+
+    /// The fields what are actually searched via tantivy.
+    ///
+    /// These values need to either be a fast field (ints) or TEXT.
+    search_fields: Vec<String>,
+
+    /// A set of fields to boost by a given factor.
```

Comment:
Could you comment a bit more the boost factor? For example, what is the range?

Reply:
Could probably mirror the docs from Tantivy's boost query to be honest, considering this is just a multiplier.

Reply:
Ah, thanks for the clarification. Putting a link to tantiviy's doc page would be helpful.